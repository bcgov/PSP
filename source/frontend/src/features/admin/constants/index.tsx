import { ColumnWithProps } from 'components/Table';
import { IOrganization, IOrganizationRecord } from 'interfaces';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

export const columnDefinitions: ColumnWithProps<IOrganizationRecord>[] = [
  {
    Header: 'Organization name',
    accessor: 'name',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Short name',
    accessor: 'code',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Description',
    accessor: 'description',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Parent Organization',
    accessor: 'parent',
    align: 'left',
    clickable: true,
    Cell: (props: CellProps<IOrganization>) => {
      return (
        <Link to={`/admin/organization/${props.row.original.parentId}`}>
          {props.row.original.parent}
        </Link>
      );
    },
  },
];
