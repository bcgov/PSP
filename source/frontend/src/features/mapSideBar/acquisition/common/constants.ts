export const cannotEditMessage =
  'This section of the file is in a non-editable state. Contact a System administrator to change it to an editable state';
