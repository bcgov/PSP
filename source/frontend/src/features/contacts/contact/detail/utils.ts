import { ContactInfoField } from 'features/contacts/interfaces';
import { Dictionary } from 'interfaces/Dictionary';
import { IContactAddress, IContactMethod } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export function getContactInfo(
  validTypes: Dictionary<string>,
  contactMethods?: IContactMethod[],
): ContactInfoField[] {
  if (contactMethods === undefined) {
    return [];
  }
  // Get only the valid types
  let filteredFields = contactMethods.reduce(
    (accumulator: ContactInfoField[], method: IContactMethod) => {
      if (Object.keys(validTypes).includes(method.contactMethodType.id)) {
        accumulator.push({
          info: method.value,
          label: validTypes[method.contactMethodType.id],
        });
      }
      return accumulator;
    },
    [],
  );

  // Sort according to the dictionary order
  return filteredFields.sort((a, b) => {
    return Object.values(validTypes).indexOf(a.label) - Object.values(validTypes).indexOf(b.label);
  });
}

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
