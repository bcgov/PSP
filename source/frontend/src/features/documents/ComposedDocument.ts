import { DocumentRelationshipType } from 'constants/documentRelationshipType';
import {
  Api_Document,
  Api_DocumentMetadataUpdate,
  Api_DocumentRelationship,
  Api_DocumentType,
  Api_DocumentUpdateRequest,
  Api_DocumentUploadRequest,
} from 'models/api/Document';
import {
  Api_Storage_DocumentDetail,
  Api_Storage_DocumentMetadata,
  Api_Storage_DocumentTypeMetadataType,
} from 'models/api/DocumentStorage';
import Api_TypeCode from 'models/api/TypeCode';

export interface ComposedDocument {
  mayanMetadata?: Api_Storage_DocumentMetadata[];
  pimsDocument?: Api_Document;
  documentDetail?: Api_Storage_DocumentDetail;
  mayanFileId?: number;
}

export class DocumentRow {
  id?: number;
  mayanDocumentId?: number;
  documentType?: Api_DocumentType;
  statusTypeCode?: Api_TypeCode<string>;
  fileName?: string;
  isFileAvailable?: boolean;
  appCreateTimestamp?: string;
  appCreateUserid?: string;
  relationshipId?: number;
  relationshipType?: DocumentRelationshipType;
  parentId?: number;

  public static fromApi(relationship: Api_DocumentRelationship): DocumentRow {
    const row: DocumentRow = new DocumentRow();
    row.id = relationship.document?.id;
    row.documentType = relationship.document?.documentType;
    row.mayanDocumentId = relationship.document?.mayanDocumentId;
    row.statusTypeCode = relationship.document?.statusTypeCode;
    row.fileName = relationship.document?.fileName;
    row.appCreateTimestamp = relationship.document?.appCreateTimestamp;
    row.appCreateUserid = relationship.document?.appCreateUserid;

    row.relationshipId = relationship.id;
    row.relationshipType = relationship.relationshipType;
    row.parentId = relationship.parentId;
    return row;
  }

  public static toApi(document: DocumentRow): Api_DocumentRelationship {
    return {
      id: document.relationshipId,
      relationshipType: document.relationshipType,
      parentId: document.parentId,
      document: {
        id: document.id,
        mayanDocumentId: document.mayanDocumentId,
        documentType: document.documentType,
        statusTypeCode: document.statusTypeCode,
        fileName: document.fileName,
      },
    };
  }
}

export class DocumentUploadFormData {
  public documentTypeId: string;
  public documentStatusCode: string;
  public documentMetadata: Record<string, string>;

  public constructor(
    initialStatus: string,
    documentType: string,
    metadata: Api_Storage_DocumentTypeMetadataType[],
  ) {
    this.documentStatusCode = initialStatus;
    this.documentTypeId = documentType;
    this.documentMetadata = {};

    metadata.forEach(metaType => {
      this.documentMetadata[metaType.metadata_type?.id?.toString() || '-'] = '';
    });
  }

  public toRequestApi(file: File, documentType: Api_DocumentType): Api_DocumentUploadRequest {
    var metadata: Api_DocumentMetadataUpdate[] = [];
    for (const key in this.documentMetadata) {
      const value = this.documentMetadata[key];
      metadata.push({
        metadataTypeId: Number(key),
        value: value,
      });
    }

    return {
      documentType: documentType,
      documentStatusCode: this.documentStatusCode,
      file: file,
      documentMetadata: metadata,
    };
  }
}

export class DocumentUpdateFormData {
  public documentId: number;
  public mayanDocumentId: number;
  public documentStatusCode: string;
  public documentMetadata: Record<string, string>;

  public static fromApi(
    composedDocument: ComposedDocument,
    metadataTypes: Api_Storage_DocumentTypeMetadataType[],
  ): DocumentUpdateFormData {
    var model = new DocumentUpdateFormData();
    model.documentId = composedDocument.pimsDocument?.id || 0;
    model.mayanDocumentId = composedDocument.pimsDocument?.mayanDocumentId || 0;
    model.documentStatusCode = composedDocument.pimsDocument?.statusTypeCode?.id?.toString() || '';
    model.documentMetadata = {};
    metadataTypes.forEach(metaType => {
      var foundMetadata = composedDocument.mayanMetadata?.find(
        currentMeta => currentMeta.metadata_type.id === metaType.metadata_type?.id,
      );
      model.documentMetadata[metaType.metadata_type?.id?.toString() || '-'] =
        foundMetadata?.value ?? '';
    });
    return model;
  }

  public toRequestApi(): Api_DocumentUpdateRequest {
    var metadata: Api_DocumentMetadataUpdate[] = [];

    for (const key in this.documentMetadata) {
      const value = this.documentMetadata[key];
      const metadataTypeId = Number(key);
      metadata.push({
        value: value,
        metadataTypeId: metadataTypeId,
      });
    }

    return {
      documentId: this.documentId,
      mayanDocumentId: this.mayanDocumentId,
      documentStatusCode: this.documentStatusCode,
      documentMetadata: metadata,
    };
  }

  private constructor() {
    this.documentId = -1;
    this.mayanDocumentId = -1;
    this.documentStatusCode = '';
    this.documentMetadata = {};
  }
}
