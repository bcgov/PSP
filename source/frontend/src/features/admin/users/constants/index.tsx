import { ReactComponent as Active } from 'assets/images/active.svg';
import { ReactComponent as Inactive } from 'assets/images/inactive.svg';
import { ColumnWithProps } from 'components/Table';
import { DateTimeCell } from 'components/Table/DateCell';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';
import { stringToFragment } from 'utils';

import { RowActions } from '../components/RowActions';
import { FormUser } from '../models';

export const getUserColumns = (refresh: () => void): ColumnWithProps<FormUser>[] => [
  {
    Header: 'Active',
    accessor: 'isDisabled',
    align: 'left',
    clickable: true,
    sortable: true,
    width: 60,
    Cell: (props: CellProps<FormUser>) =>
      props.row.original.isDisabled ? <Inactive /> : <Active />,
  },
  {
    Header: 'IDIR/BCeID',
    accessor: 'businessIdentifierValue',
    align: 'left',
    minWidth: 150,
    Cell: (props: CellProps<FormUser>) => {
      return (
        <Link to={`/admin/user/${props.row.original.id}`}>
          {props.row.original.businessIdentifierValue}
        </Link>
      );
    },
    sortable: true,
  },
  {
    Header: 'First name',
    accessor: 'firstName',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Last name',
    accessor: 'surname',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Email',
    accessor: 'email',
    align: 'left',
    clickable: true,
    sortable: true,
    minWidth: 200,
  },
  {
    Header: 'Position',
    accessor: 'position',
    align: 'left',
    clickable: true,
    sortable: true,
  },

  {
    Header: 'Roles',
    accessor: 'roles',
    align: 'left',
    clickable: true,
    minWidth: 200,
    Cell: (props: CellProps<FormUser>) => {
      const rolesString = props.row?.original?.roles?.map(userRole => userRole?.name);
      return stringToFragment(rolesString?.join(', '));
    },
  },
  {
    Header: 'MoTI region(s)',
    accessor: 'regions',
    align: 'left',
    clickable: true,
    minWidth: 200,
    Cell: (props: CellProps<FormUser>) => {
      const regionsString = props.row?.original?.regions?.map(
        regionUser => regionUser?.description,
      );
      return stringToFragment(regionsString?.join(', '));
    },
  },
  {
    Header: 'Last login',
    accessor: 'lastLogin',
    align: 'left',
    clickable: true,
    minWidth: 100,
    Cell: DateTimeCell,
  },
  {
    Header: ' ',
    Cell: (props: CellProps<FormUser>) => {
      return <RowActions {...props} refresh={refresh} />;
    },
    width: 75,
  },
];
