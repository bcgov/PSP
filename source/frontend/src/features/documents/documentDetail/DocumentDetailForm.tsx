import { Formik, FormikProps } from 'formik';
import { ChangeEvent, useCallback, useEffect, useState } from 'react';
import { useMemo } from 'react';
import { Col, Row } from 'react-bootstrap';
import styled from 'styled-components';

import { Button } from '@/components/common/buttons/Button';
import { Select, SelectOption } from '@/components/common/form';
import LoadingBackdrop from '@/components/common/LoadingBackdrop';
import { Section } from '@/components/common/Section/Section';
import { SectionField } from '@/components/common/Section/SectionField';
import TooltipIcon from '@/components/common/TooltipIcon';
import * as API from '@/constants/API';
import Claims from '@/constants/claims';
import useKeycloakWrapper from '@/hooks/useKeycloakWrapper';
import useLookupCodeHelpers from '@/hooks/useLookupCodeHelpers';
import { ApiGen_CodeTypes_DocumentRelationType } from '@/models/api/generated/ApiGen_CodeTypes_DocumentRelationType';
import { ApiGen_Concepts_DocumentType } from '@/models/api/generated/ApiGen_Concepts_DocumentType';
import { ApiGen_Mayan_DocumentTypeMetadataType } from '@/models/api/generated/ApiGen_Mayan_DocumentTypeMetadataType';
import { ApiGen_Requests_DocumentUpdateRequest } from '@/models/api/generated/ApiGen_Requests_DocumentUpdateRequest';
import { exists } from '@/utils/utils';

import { StyledH3, StyledScrollable } from '../commonStyles';
import { ComposedDocument, DocumentUpdateFormData } from '../ComposedDocument';
import { DocumentMetadataView } from '../DocumentMetadataView';
import { StyledContainer } from '../list/styles';
import DocumentDetailHeader from './DocumentDetailHeader';
import { DocumentUpdateFormDataYupSchema } from './DocumentUpdateFormDataYupSchema';

export interface IDocumentDetailFormProps {
  formikRef: React.RefObject<FormikProps<DocumentUpdateFormData>>;
  document: ComposedDocument;
  isLoading: boolean;
  documentTypes: ApiGen_Concepts_DocumentType[];
  mayanMetadataTypes: ApiGen_Mayan_DocumentTypeMetadataType[];
  relationshipType: ApiGen_CodeTypes_DocumentRelationType;
  documentTypeUpdated: boolean;
  onDocumentTypeChange: (changeEvent: React.ChangeEvent<HTMLInputElement>) => void;
  onUpdate: (updateRequest: ApiGen_Requests_DocumentUpdateRequest) => void;
  onCancel: () => void;
}

/**
 * Component that provides functionality to see document information. Can be embedded as a widget.
 */
export const DocumentDetailForm: React.FunctionComponent<
  React.PropsWithChildren<IDocumentDetailFormProps>
> = props => {
  const { hasClaim } = useKeycloakWrapper();

  const { getOptionsByType } = useLookupCodeHelpers();
  const documentStatusTypes = getOptionsByType(API.DOCUMENT_STATUS_TYPES);
  const initialFormState = DocumentUpdateFormData.fromApi(props.document, props.mayanMetadataTypes);

  const documentTypeOptions = useMemo(
    () =>
      props.documentTypes.map<SelectOption>(x => {
        return { label: x.documentTypeDescription || '', value: x.id?.toString() || '' };
      }),
    [props.documentTypes],
  );

  const [documentTypePurpose, setDocumentTypePurpose] = useState(null);

  const matchDocumentType = useCallback(
    (documentTypeId: number) => {
      const purpose = props.documentTypes.find(x => x.id === documentTypeId)?.documentTypePurpose;

      setDocumentTypePurpose(purpose);
    },
    [props.documentTypes],
  );

  useEffect(() => {
    const documentTypeId = Number(props.formikRef.current?.values?.documentTypeId);
    matchDocumentType(documentTypeId);
  }, [matchDocumentType, props.formikRef]);

  const onDocumentTypeChange = useCallback(
    async (changeEvent: ChangeEvent<HTMLInputElement>) => {
      if (changeEvent.target.value) {
        const documentTypeId = Number(changeEvent.target.value);
        matchDocumentType(documentTypeId);
      }

      props.onDocumentTypeChange(changeEvent);
    },
    [matchDocumentType, props],
  );

  return (
    <StyledContainer>
      <LoadingBackdrop show={props.isLoading} />
      {hasClaim(Claims.DOCUMENT_VIEW) && (
        <>
          <DocumentDetailHeader document={props.document} />
          <Section
            noPadding
            header={
              <>
                Document Information
                <TooltipIcon
                  toolTipId="documentInfoToolTip"
                  innerClassName="documentInfoToolTip"
                  toolTip="Information you provided here will be searchable"
                />{' '}
              </>
            }
          >
            <StyledScrollable>
              <Formik<DocumentUpdateFormData>
                innerRef={props.formikRef}
                initialValues={initialFormState}
                validationSchema={DocumentUpdateFormDataYupSchema}
                onSubmit={async (values: DocumentUpdateFormData, { setSubmitting }) => {
                  if (
                    props.document?.pimsDocumentRelationship?.id &&
                    values.documentStatusCode !== undefined
                  ) {
                    const request = DocumentUpdateFormData.toRequestApi(values);
                    await props.onUpdate(request);
                    setSubmitting(false);
                  } else {
                    console.error('Selected document status is not valid');
                  }
                }}
              >
                {formikProps => (
                  <>
                    <SectionField label="Document type" labelWidth="4" required>
                      <Select
                        className="mb-0"
                        placeholder={
                          documentTypeOptions.length > 1 ? 'Select Document type' : undefined
                        }
                        field={'documentTypeId'}
                        options={documentTypeOptions}
                        onChange={onDocumentTypeChange}
                        disabled={
                          documentTypeOptions.length === 1 ||
                          props.relationshipType === ApiGen_CodeTypes_DocumentRelationType.Templates
                        }
                      />
                    </SectionField>
                    {exists(documentTypePurpose) && (
                      <SectionField label={null}>
                        <StyledPurposeText>{documentTypePurpose}</StyledPurposeText>
                      </SectionField>
                    )}
                    <SectionField label="Status" labelWidth="4">
                      <Select field="documentStatusCode" options={documentStatusTypes} />
                    </SectionField>
                    {props.documentTypeUpdated && (
                      <StyledDiv>
                        <em>
                          Some associated metadata may be lost if the document type is changed.
                        </em>
                      </StyledDiv>
                    )}
                    <StyledH3>Details</StyledH3>
                    <DocumentMetadataView
                      mayanMetadata={props.mayanMetadataTypes}
                      values={formikProps.values}
                      errors={formikProps.errors}
                    />
                    <div className="pt-5">Do you want to proceed?</div>
                    <hr />
                    <Row className="justify-content-end pt-4">
                      <Col xs="auto">
                        <Button
                          variant="secondary"
                          type="button"
                          onClick={props.onCancel}
                          className="px-5"
                        >
                          No
                        </Button>
                      </Col>
                      <Col xs="auto">
                        <Button type="submit" onClick={formikProps.submitForm} className="px-5">
                          Yes
                        </Button>
                      </Col>
                    </Row>
                  </>
                )}
              </Formik>
            </StyledScrollable>
          </Section>
        </>
      )}
    </StyledContainer>
  );
};

export const StyledDiv = styled.div`
  margin-bottom: 1rem;
`;

const StyledPurposeText = styled.div`
  color: black;
  font-style: italic;
`;
