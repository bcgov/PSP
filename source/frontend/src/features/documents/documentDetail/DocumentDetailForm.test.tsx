import { Claims } from 'constants/claims';
import { createMemoryHistory } from 'history';
import { mockLookups } from 'mocks/mockLookups';
import { Api_DocumentType } from 'models/api/Document';
import { Api_Storage_DocumentMetadata, Api_Storage_MetadataType } from 'models/api/DocumentStorage';
import { lookupCodesSlice } from 'store/slices/lookupCodes';
import { mockKeycloak, render, RenderOptions } from 'utils/test-utils';

import { ComposedDocument } from '../ComposedDocument';
import { DocumentDetailForm } from './DocumentDetailForm';

// mock auth library
jest.mock('@react-keycloak/web');

const history = createMemoryHistory();

const documentTypes: Api_DocumentType[] = [
  {
    id: 1,
    documentType: 'BC Assessment Search',
    mayanId: 17,
  },
  {
    id: 2,
    documentType: 'Privy Council',
    mayanId: 7,
  },
];
const metadataTypes: Api_Storage_MetadataType[] = [
  { id: 1, label: 'Tag Foo', name: 'tag-foo' },
  { id: 2, label: 'Tag Bar', name: 'tag-bar' },
];
const documentTypeMetadata: Api_Storage_DocumentMetadata[] = [
  {
    document: {
      label: '',
      datetime_created: '2022-07-27T16:06:42.42',
      description: '',
      file_latest: {
        id: 2,
        document_id: 1,
        comment: '',
        encoding: '',
        fileName: '',
        mimetype: '',
        size: 12,
        timeStamp: '',
      },
      id: 1,
      document_type: { id: 1, label: 'BC Assessment Search' },
    },
    id: 1,
    metadata_type: metadataTypes[0],
    url: '',
    value: 'Tag1234',
  },
];
const mockDocument: ComposedDocument = {
  mayanMetadata: documentTypeMetadata,
  pimsDocumentRelationship: {
    id: 1,
    document: {
      mayanDocumentId: 15,
      documentType: documentTypes[0],
      statusTypeCode: { id: 'AMEND', description: 'Amended' },
      fileName: 'NewFile.doc',
    },
    parentId: undefined,
    relationshipType: undefined,
  },
  mayanFileId: 2,
};
describe('DocumentDetailForm component', () => {
  // render component under test
  const setup = (renderOptions: RenderOptions) => {
    const utils = render(
      <DocumentDetailForm
        onUpdate={jest.fn()}
        document={mockDocument}
        isLoading={false}
        mayanMetadataTypes={documentTypeMetadata}
        onCancel={jest.fn()}
        formikRef={{ current: { submitForm: jest.fn() } } as any}
      />,
      {
        ...renderOptions,
        store: {
          [lookupCodesSlice.name]: { lookupCodes: mockLookups },
        },
        history,
      },
    );

    return {
      ...utils,
      useMockAuthentication: true,
    };
  };

  beforeEach(() => {
    mockKeycloak({ claims: [Claims.DOCUMENT_VIEW, Claims.DOCUMENT_EDIT] });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders as expected', () => {
    setup({});
    expect(document.body).toMatchSnapshot();
  });

  it('renders the file name', () => {
    const { getAllByText } = setup({});
    const textarea = getAllByText('NewFile.doc')[0];

    expect(textarea).toBeVisible();
  });

  it('renders the document type', () => {
    const { getAllByText } = setup({});
    const textarea = getAllByText('BC Assessment Search')[0];

    expect(textarea).toBeVisible();
  });

  it('displays field for metadata types', async () => {
    const { getAllByDisplayValue } = setup({});
    const textarea = getAllByDisplayValue('Tag1234')[0];

    expect(textarea).toBeVisible();
  });
});
