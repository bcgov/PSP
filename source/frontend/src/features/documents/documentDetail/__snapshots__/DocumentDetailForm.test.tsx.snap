// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentDetailForm component renders as expected 1`] = `
.c9.btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.4rem 1.2rem;
  min-height: 3rem;
  border: 0.2rem solid transparent;
  border-radius: 0.4rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  font-family: 'BCSans','Noto Sans',Verdana,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  cursor: pointer;
}

.c9.btn:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c9.btn:focus {
  outline-width: 0.4rem;
  outline-style: solid;
  outline-offset: 1px;
  box-shadow: none;
}

.c9.btn.btn-primary {
  border: none;
}

.c9.btn.btn-secondary {
  background: none;
}

.c9.btn.btn-info {
  border: none;
  background: none;
  padding-left: 0.6rem;
  padding-right: 0.6rem;
}

.c9.btn.btn-info:hover,
.c9.btn.btn-info:active,
.c9.btn.btn-info:focus {
  background: none;
}

.c9.btn.btn-light {
  border: none;
}

.c9.btn.btn-dark {
  border: none;
}

.c9.btn.btn-link {
  font-size: 1.6rem;
  font-weight: 400;
  background: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 2.5rem;
  line-height: 3rem;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  text-align: left;
  padding: 0;
}

.c9.btn.btn-link:hover,
.c9.btn.btn-link:active,
.c9.btn.btn-link:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  border: none;
  background: none;
  box-shadow: none;
  outline: none;
}

.c9.btn.btn-link:disabled,
.c9.btn.btn-link.disabled {
  background: none;
  pointer-events: none;
}

.c9.btn:disabled,
.c9.btn:disabled:hover {
  box-shadow: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.65;
}

.c9.Button .Button__icon {
  margin-right: 1.6rem;
}

.c9.Button--icon-only:focus {
  outline: none;
}

.c9.Button--icon-only .Button__icon {
  margin-right: 0;
}

.c2.required::before {
  content: '*';
  position: absolute;
  top: 0.75rem;
  left: 0rem;
}

.c1 {
  font-weight: bold;
}

.c6 {
  width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow-y: auto;
}

.c3 {
  padding: 1rem;
}

.c5 {
  font-weight: 700;
}

.c8 {
  font-weight: 700;
  font-size: 1.7rem;
  margin-bottom: 1rem;
  text-align: left;
  padding-top: 1rem;
  border-bottom: solid 0.1rem;
}

.c4 {
  text-align: left !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  overflow-x: hidden;
  max-height: 50rem;
}

.c0 {
  padding: 1rem;
}

<body>
  <div>
    <div
      class="Toastify"
    />
    <div />
    <div
      class="c0"
    >
      <div
        class="pb-2 row"
      >
        <div
          class="pr-0 text-left col-4"
        >
          <label
            class="c1"
          >
            Document type
            :
          </label>
        </div>
        <div
          class="c2 text-left col"
        >
          BC Assessment Search
        </div>
      </div>
      <div
        class="pb-3 row"
      >
        <div
          class="pr-0 text-left col-4"
        >
          <label
            class="c1"
          >
            File name
            :
          </label>
        </div>
        <div
          class="c2 text-left col"
        >
          <div
            class="row"
          >
            <div
              class="col-auto"
            >
              NewFile.doc
            </div>
            <div
              class="col-auto"
            >
              <span
                class="tooltip-icon"
                data-testid="tooltip-icon-document-not-available-tooltip"
                id="document-not-available-tooltip"
              >
                <svg
                  class="tooltip-icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c3"
      >
        <div
          class="pb-3 row"
        >
          <div
            class="text-left col"
          >
            <div
              class="c4"
            >
              <h2
                class="c5"
              >
                Document information
              </h2>
              <span
                class="tooltip-icon"
                data-testid="tooltip-icon-documentInfoToolTip"
                id="documentInfoToolTip"
              >
                <svg
                  class="tooltip-icon documentInfoToolTip"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="c6 c7"
        >
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-4"
            >
              <label
                class="c1"
              >
                Status
                :
              </label>
            </div>
            <div
              class="c2 text-left col"
            >
              <div
                class="form-group"
              >
                <select
                  class="form-select form-control"
                  id="input-documentStatusCode"
                  name="documentStatusCode"
                >
                  <option
                    class="option"
                    value="AMENDD"
                  >
                    Amended
                  </option>
                  <option
                    class="option"
                    value="APPROVD"
                  >
                    Approved
                  </option>
                  <option
                    class="option"
                    value="CNCLD"
                  >
                    Cancelled
                  </option>
                  <option
                    class="option"
                    value="DRAFT"
                  >
                    Draft
                  </option>
                  <option
                    class="option"
                    value="FINAL"
                  >
                    Final
                  </option>
                  <option
                    class="option"
                    value="NONE"
                  >
                    None
                  </option>
                  <option
                    class="option"
                    value="RGSTRD"
                  >
                    Registered
                  </option>
                  <option
                    class="option"
                    value="SENT"
                  >
                    Sent
                  </option>
                  <option
                    class="option"
                    value="SIGND"
                  >
                    Signed
                  </option>
                  <option
                    class="option"
                    value="UNREGD"
                  >
                    Unregistered
                  </option>
                </select>
              </div>
            </div>
          </div>
          <h3
            class="c8"
          >
            Details
          </h3>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-4"
            >
              <label
                class="c1"
              >
                Tag Foo
                :
              </label>
            </div>
            <div
              class="c2 text-left col"
            >
              <div
                class="input form-group"
              >
                <input
                  class="form-control"
                  data-testid="metadata-input-tag-foo"
                  id="input-documentMetadata.1"
                  name="documentMetadata.1"
                  title="Tag1234"
                  value="Tag1234"
                />
              </div>
            </div>
          </div>
          <div
            style="color: red;"
          />
          <div
            class="justify-content-end pt-4 row"
          >
            <div
              class="col-auto"
            >
              <button
                class="c9 Button btn btn-secondary"
                type="button"
              >
                <div
                  class="Button__value"
                >
                  Cancel
                </div>
              </button>
            </div>
            <div
              class="col-auto"
            >
              <button
                class="c9 Button btn btn-primary"
                type="submit"
              >
                <div
                  class="Button__value"
                >
                  Save
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
